orca-clouddriver/src/main/groovy/com/netflix/spinnaker/orca/clouddriver/tasks/image/ImageTagger.java

```diff
@@ -102,6 +108,14 @@ protected Collection findImages(Collection<String> imageNames, Collection<String
       foundImages.add(objectMapper.convertValue(matchedImage, matchedImageType));
     }
 
+    if (foundImages.size() < upstreamImageIds.size()) {
+      throw new ImageNotFound(
+        format("Only found %d images for names %s but %d upstreamImageIds",
```

Comment:
nit: this could show up in the UI, right? If so, maybe something a bit digesteable for the average user would be better, e.g.
`Only found %d images to tag but %d were specified (%s)`

Reply:
This is a special exception type that results in the message being logged and the task requeued. It isn’t a failure and shouldn’t surface to deck or execution context. Will clarify the language for the sake of our logs however.

Reply:
This is a special exception type that results in the message being logged and the task requeued. It isn’t a failure and shouldn’t surface to deck or execution context. Will clarify the language for the sake of our logs however.

Reply:
i see, I thought that the last one of these would get logged together with the timeout status/exception, but now i can't find that code